src/layman/layer/filesystem/input_style.py

```diff
@@ -66,6 +73,44 @@ def save_layer_file(workspace, layername, style_file, style_type):
                 out.write(style_file.read())
 
 
+def get_mapping_from_external_image_list(external_images):
+    return {path: os.path.join(filesystem.EXTERNAL_IMAGES_DIR, f'image_{idx}{os.path.splitext(path)[1]}') for idx, path in
+            enumerate(external_images)}
+
+
+def save_layer_files(workspace, layername, style_file, style_type, style_files, external_image_paths, ):
+    delete_layer(workspace, layername)
+
+    if style_file:
+        ensure_layer_input_style_dir(workspace, layername)
+        style_path_clear = get_file_path(workspace, layername, with_extension=False)
+        style_path = style_path_clear + '.' + style_type.extension
+
+        if isinstance(style_file, FileStorage):
+            xml = style_file.read()
+            style_file.seek(0)
+            qml_etree = etree.fromstring(xml)
+        else:
+            qml_etree = etree.parse(style_file)
+
+        if external_image_paths:
+            ext_image_mapping = get_mapping_from_external_image_list(external_image_paths)
+            input_style_dir = get_layer_input_style_dir(workspace, layername)
+            external_images = [ext_image for ext_image in style_files if
+                               ext_image != style_file and ext_image.filename in ext_image_mapping]
+            common_input_file.save_files(external_images, ext_image_mapping, prefix=input_style_dir)
+
+            for xpath, attr_name in EXTERNAL_IMAGES_XPATHS:
```

Comment:
I suggest to move this for cycle into `adjust_qml_external_image_paths(xml_etree, image_mapping)` method and call this method from here. Also, I would call this method only if in case of QML file, otherwise raise unsupported exception.